### PR TITLE
D3ASIM-1328: Added deserialization functionality on d3a to handle inc…

### PR DESCRIPTION
**Pull request by spyrostz**
…oming area and settings events via Redis job. Added unit tests.

### CODE REVIEW

File: src/d3a/d3a_core/simulation.py [hunk 1]
```diff
@@ -94,7 +96,12 @@ def __init__(self, setup_module_name: str, simulation_config: SimulationConfig =
 
         self._load_setup_module()
         self._init(**self.initial_params)
+        # TODO: Deprecated event mechanism since D3ASIM-1328 is about to be merged.
+        # Candidate for deletion.
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/simulation.py]: Dont we need that when running locally setup files that contain events?

File: src/d3a/models/area/event_deserializer.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                  'transfer_fee_pct': event.get("iaa_fee", None),
+                  'market_maker_rate': market_maker_rate}
+        output_events.append(ConfigEvents(event_time, params))
+    return output_events
+
+
+def deserialize_events_to_areas(events, root_area):
+    if not events:
+        return
+    event_list = generate_settings_events(events.get("settings_events", [])) \
```
Comment (hannesdiedrich) [inline src/d3a/models/area/event_deserializer.py]: Could you please rename `event_list` to `settings_event_list`  for better understandability? It took me a while. ;-)

File: src/d3a/models/area/event_deserializer.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    area.set_events(event_list)
+
+    if not area.children:
+        return
+    for child in area.children:
+        assign_events_to_areas(child, area_events, settings_events)
+
+
+def generate_settings_events(settings_events):
+    output_events = []
```
Comment (hannesdiedrich) [inline src/d3a/models/area/event_deserializer.py]: And maybe `output_events` to `settings_event_list`, too? 
Very minor of corse! Thanks a lot!

File: src/d3a/d3a_core/simulation.py (hunk 1 above)
Comment (spyrostz) [reply src/d3a/d3a_core/simulation.py]: No, this is from the event mechanism that Brainbot had introduced before taking over the project. Unfortunately it was partially implemented, that's why we introduced the new event mechanism. As far as I understand, the old event mechanism is not used at all anymore (apart from, maybe, unit tests) and we can safely remove it.

File: src/d3a/models/area/event_deserializer.py (hunk 3 above)
Comment (spyrostz) [reply src/d3a/models/area/event_deserializer.py]: I agree the naming is not clear, will do , thanks!

File: src/d3a/d3a_core/simulation.py (hunk 1 above)
Comment (hannesdiedrich) [reply src/d3a/d3a_core/simulation.py]: Ok, understood. So please go ahead and delete it. Thanks a lot!

File: src/d3a/models/area/event_deserializer.py (hunk 2 above)
Comment (spyrostz) [reply src/d3a/models/area/event_deserializer.py]: You are right again, will rename this. Thanks!

**Pull request merged**